src/framework/components/particle-system/component.js

```diff
@@ -696,27 +755,43 @@ Object.assign(pc, function () {
                 this.addModelToLayers();
             }
 
-            this.system.app.scene.on("set:layers", this.onLayersChanged, this);
-            if (this.system.app.scene.layers) {
-                this.system.app.scene.layers.on("add", this.onLayerAdded, this);
-                this.system.app.scene.layers.on("remove", this.onLayerRemoved, this);
+            var scene = system.app.scene;
+            scene.on("set:layers", this.onLayersChanged, this);
+            if (scene.layers) {
+                scene.layers.on("add", this.onLayerAdded, this);
+                scene.layers.on("remove", this.onLayerRemoved, this);
             }
 
             if (this.enabled && this.entity.enabled && data.depthSoftening) {
                 this._requestDepth();
             }
+
+            // Add to list of enabled particle systems to easily update them
+            system._enabledParticleSystems.push(this);
         },
 
         onDisable: function () {
-            this.system.app.scene.off("set:layers", this.onLayersChanged, this);
-            if (this.system.app.scene.layers) {
-                this.system.app.scene.layers.off("add", this.onLayerAdded, this);
-                this.system.app.scene.layers.off("remove", this.onLayerRemoved, this);
+            var data = this.data;
+            var system = this.system;
+
+            // Remove from enabled particle systems array
+            var idx = system._enabledParticleSystems.indexOf(this);
+            if (idx > -1) {
+                system._enabledParticleSystems.splice(idx, 1);
```

Comment:
Minor: I would prefer this to be done through a function/property as `_enabledParticleSystems` is considered private to the `system`. But if we do this elsewhere, that's fine.